api/v1alpha1/ionoscloudcluster_types_test.go

```diff
@@ -86,14 +86,15 @@ var _ = Describe("IonosCloudCluster", func() {
 			Expect(k8sClient.Create(context.Background(), cluster)).
 				Should(MatchError(ContainSubstring("credentialsRef.name must be provided")))
 		})
-		It("should not allow creating clusters with empty location when ControlPlaneEndpoint host is set", func() {
+		It("should not allow creating clusters with empty location when loadBalancerProviderRef is set", func() {
 			cluster := defaultCluster()
 			cluster.Spec.Location = ""
 			Expect(k8sClient.Create(context.Background(), cluster)).
-				Should(MatchError(ContainSubstring("location is required when controlPlaneEndpoint.host is set")))
+				Should(MatchError(ContainSubstring("location is required when loadBalancerProviderRef is set")))
 		})
 		It("should allow creating clusters with empty location when ControlPlaneEndpoint host is not set", func() {
```

Comment:
`ControlPlaneEndpoint` leftover

Reply:
no, that's correct

Reply:
Hm, I disagree. This test case and the previous one belong together.

The previous one is `should not allow creating clusters with empty location when loadBalancerProviderRef is set` and this one should be `should allow creating clusters with empty location when loadBalancerProviderRef is not set`.

Also no need for `cluster.Spec.ControlPlaneEndpoint.Host = ""` below then.

Reply:
the host part is important because that's needed for kamaji integration so we must test that it works